Cython/Shadow.py

```diff
@@ -572,12 +574,45 @@ def __getattr__(self, item):
             raise ex
 
 
+class CythonPyCImporter:
+    """
+    A meta-path import hook to map 'cython.cimports.*' imports to plain Python module imports.
+
+    Maps 'import cython.cimports.mypackage.mymodule' to 'import mypackage.mymodule'.
+    This allows using pxd file based cimports in compiled code that work with the
+    corresponding Python modules (compiled or not compiled) when run by normal Python.
+    """
+    @staticmethod
+    def find_spec(name, import_path, target_module):
+        if not name.startswith('cython.cimports.'):
+            return None
+
+        # Try to import a Python module of the same name (without prefix) instead.
+        module_name = name[len('cython.cimports.'):]
+
+        import sys
+        if module_name in sys.modules:
+            module = sys.modules[module_name]
+            sys.modules[name] = module
+            return module.__spec__
+
+        from importlib.util import find_spec
+        spec = find_spec(module_name)
+        return spec
+
+
 import math, sys
+
 sys.modules['cython.parallel'] = CythonDotParallel()
 sys.modules['cython.cimports'] = CythonCImports('cython.cimports')
 sys.modules['cython.cimports.libc'] = CythonCImports('cython.cimports.libc')
```

Comment:
Should these now be removed? Or are they still a separate mechanism for something?

Reply:
There isn't a `libc` Python module, so I'm not sure this can just be removed. You can write `from cython.cimports.libc import math`, for example, and it won't hit the registered `libc.math` module (I think).